mkosi/__init__.py

```diff
@@ -998,11 +1001,14 @@ def build_uki(
 
     cmdline += state.config.kernel_command_line
 
+    if not (arch := state.config.architecture.to_efi()):
+        die(f"Architecture {state.config.architecture} does not support UEFI")
+
```

Comment:
Is there a need to have this check above the comment? Having it closer to the use in the definition of `stub` would make it easier to follow where `arch` comes from there.